src/test/scala/com/exasol/cloudetl/kafka/consumer/KafkaRecordConsumerTest.scala

```diff
@@ -0,0 +1,132 @@
+package com.exasol.cloudetl.kafka
+
+import java.time.Duration
+import java.util.{Map => JMap}
+import java.util.Arrays
+import java.util.Collections
+
+import com.exasol.ExaIterator
+import com.exasol.cloudetl.kafka.consumer.KafkaRecordConsumer
+import com.exasol.cloudetl.kafka.deserialization.FieldSpecification
+import com.exasol.cloudetl.kafka.deserialization.RecordKey
+import com.exasol.cloudetl.kafka.deserialization.RecordValue
+
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.ConsumerRecords
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.TopicPartition
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.mockito.Mockito.when
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatestplus.mockito.MockitoSugar
+
+class KafkaRecordConsumerTest extends AnyFunSuite with BeforeAndAfterEach with MockitoSugar {
+
+  private[this] val topicName = "topicName"
+  private[this] val topicPartition = new TopicPartition(topicName, 0)
+  private[this] val defaultProperties = Map(
+    "TOPIC_NAME" -> topicName,
+    "MIN_RECORDS_PER_RUN" -> "2",
+    "MAX_RECORDS_PER_RUN" -> "4",
+    "RECORD_KEY_FORMAT" -> "string",
+    "RECORD_VALUE_FORMAT" -> "string"
+  )
+  private[this] val defaultTimeout = Duration.ofMillis(30000)
+  private[this] val defaultEndOffset = 4L
+  private[this] val emptyConsumerRecords =
+    new ConsumerRecords[FieldType, FieldType](Collections.emptyMap())
+
+  type FieldType = Map[FieldSpecification, Seq[Any]]
+  private[this] var iterator: ExaIterator = _
+  private[this] var consumer: KafkaConsumer[FieldType, FieldType] = _
+
+  override final def beforeEach(): Unit = {
+    iterator = mock[ExaIterator]
+    consumer = mock[StubConsumer]
+    when(consumer.endOffsets(Arrays.asList(topicPartition)))
+      .thenReturn(JMap.of(topicPartition, defaultEndOffset))
+    ()
+  }
+
+  test("emits all records using min and max record counts") {
+    when(consumer.poll(defaultTimeout))
+      .thenReturn(recordBatch(Seq(0, 1)), recordBatch(Seq(2, 3)))
+    KafkaImportChecker().assertEmitCount(4)
+  }
+
+  test("emits all records using min and max record counts with empty records") {
+    when(consumer.poll(defaultTimeout))
+      .thenReturn(recordBatch(Seq(0, 1)), emptyConsumerRecords, recordBatch(Seq(2, 3)))
+    KafkaImportChecker().assertEmitCount(2)
+  }
+
+  test("emits all records using consume_all_offsets") {
+    when(consumer.poll(defaultTimeout))
+      .thenReturn(recordBatch(Seq(0, 1)), recordBatch(Seq(2, 3)))
+    KafkaImportChecker(Map("CONSUME_ALL_OFFSETS" -> "true")).assertEmitCount(4)
```

Comment:
What would happen if we set CONSUME_ALL_OFFSETS and MAX_RECORDS_PER_RUN at the same time?

Reply:
Should we add a test case for it?

Reply:
Feels like that should probably fail and give an explanatory error message back to the user, right? 🤔

Reply:
Consuming all offset will override other two parameters, I think it should be already described on the user guide. I do not think we need to validate them for now.

Reply:
Added test to cover this case